### PR TITLE
Adds a CORS middleware that accepts all Cross-Origin Requests.

### PR DESCRIPTION
**Pull request by PradyumnaShome**
The middleware sets the suitable CORS-related headers to allow uploads.

This middleware is chained to all the upload service routes.

### CODE REVIEW

File: gateway/middleware/cors.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+func CorsMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		AllowCors(w, r)
+		next.ServeHTTP(w, r)
+	})
+}
+
+func AllowCors(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Was-CORS-Set", "Yes, CORS was set.")
```
Comment (ASankaran) [inline gateway/middleware/cors.go]: Why this header?

File: gateway/middleware/cors.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Methods", "GET, POST, PUT, PATCH, DELETE, OPTIONS, CONNECT")
+	w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization, Origin, Content-Disposition")
+	w.WriteHeader(http.StatusOK)
```
Comment (ASankaran) [inline gateway/middleware/cors.go]: You can't write headers here. The final handler might need to modify them or the http code might not be 200.

File: gateway/services/services.go [hunk 3]
```diff
@@ -40,10 +41,3 @@ func RegisterAPIs() arbor.RouteCollection {
 	Routes = append(Routes, StatRoutes...)
 	return Routes
 }
-
-func AllowCors(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Access-Control-Allow-Origin", "*")
-	w.Header().Set("Access-Control-Allow-Methods", "GET, POST, PUT, PATCH, DELETE, OPTIONS, CONNECT")
-	w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization, Origin, Content-Disposition")
-	w.WriteHeader(http.StatusOK)
```
Comment (ASankaran) [inline gateway/services/services.go]: You should be writing the header here.

File: gateway/middleware/cors.go (hunk 2 above)
Comment (PradyumnaShome) [reply gateway/middleware/cors.go]: Good point, will force push again.

File: gateway/middleware/cors.go (hunk 1 above)
Comment (PradyumnaShome) [reply gateway/middleware/cors.go]: Thought I over-wrote this already.

File: gateway/services/services.go (hunk 3 above)
Comment (PradyumnaShome) [reply gateway/services/services.go]: But I did ;)

**Pull request merged**